src/api/spec/controllers/webui/obs_factory/distributions_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+require 'webmock/rspec'
+# WARNING: If you change owner tests make sure you uncomment this line
+# and start a test backend. Some of the Owner methods
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
+
+RSpec.describe Webui::ObsFactory::DistributionsController, type: :controller, vcr: true do
+  let(:factory) { create(:project, name: 'openSUSE:Factory') }
+  let!(:factory_staging_a) { create(:project, name: 'openSUSE:Factory:Staging:A', description: '') }
+  let!(:factory_ring_bootstrap) { create(:project, name: 'openSUSE:Factory:Rings:0-Bootstrap', description: '') }
+  let(:target_package) { create(:package, name: 'target_package', project: factory) }
+  let(:source_project) { create(:project, name: 'source_project') }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+
+  describe 'GET #show' do
+    before do
+      allow(::ObsFactory::OpenqaJob).to receive(:find_all_by).and_return([:fake, :content])
+
+      get :show, params: { project: factory.name }
+    end
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:show) }
+
+    it { expect(assigns(:staging_projects).first.project).to eq(factory_staging_a) }
+
+    it { expect(assigns(:versions)[:source]).to eq(assigns(:distribution).source_version) }
+    it { expect(assigns(:versions)[:totest]).to eq(assigns(:distribution).totest_version) }
+    it { expect(assigns(:versions)[:published]).to eq(assigns(:distribution).published_version) }
+
+    it { expect(assigns(:ring_prjs).first.project).to eq(factory_ring_bootstrap) }
+    it { expect(assigns(:standard).nickname).to eq('standard') }
+    it { expect(assigns(:images).nickname).to eq('images') }
+    it { expect(assigns(:live)).to be_nil }
+    it { expect(assigns(:openqa_jobs)).to eq([:fake, :content]) }
+
+    context 'calculate_reviews' do
+      let!(:repo_checker_user) { create(:user, login: 'repo-checker') }
+      let!(:create_groups) do
+        ['opensuse-review-team', 'factory-auto', 'legal-auto', 'legal-team'].each do |title|
+          create(:group, title: title)
+        end
+      end
+      let!(:create_review_requests) do
+        ['opensuse-review-team', 'factory-auto', 'legal-auto', 'legal-team'].map do |group_name|
+          ObsFactory::Request.new(create(:review_bs_request_by_group,
+                                         reviewer: group_name,
+                                         target_project: factory.name,
+                                         target_package: target_package.name,
+                                         source_project: source_package.project.name,
+                                         source_package: source_package.name))
+        end
+        ObsFactory::Request.new(create(:review_bs_request,
```

Comment:
:thought_balloon: why do this outside the loop?

Reply:
It it a review for user, not for groups... see the first param

Reply:
So? It's just a string right? :-)

Reply:
If I have to move this inside the loop, then I have to check if the title is 'repo-checker' to create a different type of factory... does it make sense to not "repeat" the creation?

It would end up like this:
```ruby
let!(:create_groups_and_reviews) do
  ['opensuse-review-team', 'factory-auto', 'legal-auto', 'legal-team', 'repo-checker'].each do |title|
    create(:group, title: title)
    ObsFactory::Request.new(create(title == 'repo-checker' ? :review_bs_request : :review_bs_request_by_group,
                                   reviewer: title,
                                   target_project: factory.name,
                                   target_package: target_package.name,
                                   source_project: source_package.project.name,
                                   source_package: source_package.name))
  end
end
```

Do you like it more? Is not too much obfuscated?

Reply:
Hm no both in one sucks then.